src/Microsoft.TemplateEngine.Cli/TemplateResolution/TemplateListResolver.cs

```diff
@@ -351,29 +351,23 @@ private static void AddDefaultLanguageMatchingToTemplates(IReadOnlyCollection<IT
 
             foreach (ITemplateMatchInfo template in listToFilter)
             {
-                if (template.Info.Tags != null)
-                {
-                    if (template.Info.Tags.TryGetValue("language", out ICacheTag languageTag))
-                    {
-                        MatchKind matchKind;
-
-                        if (languageTag.ChoicesAndDescriptions.ContainsKey(language))
-                        {
-                            matchKind = MatchKind.Exact;
-                        }
-                        else
-                        {
-                            matchKind = MatchKind.Mismatch;
-                        }
+                MatchKind matchKind;
 
-                        // only add default language disposition when there is a language specified for the template.
-                        template.AddDisposition(new MatchInfo
-                        {
-                            Location = MatchLocation.DefaultLanguage,
-                            Kind = matchKind
-                        });
-                    }
+                if (template.Info.GetLanguages().Any(lang => lang.Equals(language, StringComparison.OrdinalIgnoreCase)))
+                {
+                    matchKind = MatchKind.Exact;
+                }
+                else
+                {
+                    matchKind = MatchKind.Mismatch;
                 }
+
```

Comment:
known bug here: disposition should not be added in case language is not defined in template.